patches/server/0731-Goat-ram-API.patch

```diff
@@ -0,0 +1,52 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Seggan <segganew@gmail.com>
+Date: Thu, 5 Aug 2021 13:10:27 -0400
+Subject: [PATCH] Goat ram API
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/animal/goat/Goat.java b/src/main/java/net/minecraft/world/entity/animal/goat/Goat.java
+index 15787afad42f9299638a1c9e57d26678805f18ee..5f019936f002f7f50246cc51643cc68f305a97d8 100644
+--- a/src/main/java/net/minecraft/world/entity/animal/goat/Goat.java
++++ b/src/main/java/net/minecraft/world/entity/animal/goat/Goat.java
+@@ -279,6 +279,14 @@ public class Goat extends Animal {
+         return new Goat.GoatPathNavigation(this, world);
+     }
+ 
++    // Paper start - Goat ram API
++    public void ram(LivingEntity entity) {
++        getBrain().setMemory(MemoryModuleType.RAM_TARGET, entity.position());
++        getBrain().eraseMemory(MemoryModuleType.RAM_COOLDOWN_TICKS);
++        getBrain().setActiveActivityIfPossible(net.minecraft.world.entity.schedule.Activity.RAM);
++    }
++    // Paper end
++
+     private static class GoatPathNavigation extends GroundPathNavigation {
+ 
+         GoatPathNavigation(Goat goat, Level world) {
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftGoat.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftGoat.java
+index ae74df5c9845ac125968a52897f4343b0f348217..48f1503b3e4309223e23610f57fe1fb2134d54d3 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftGoat.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftGoat.java
+@@ -3,6 +3,10 @@ package org.bukkit.craftbukkit.entity;
+ import org.bukkit.craftbukkit.CraftServer;
+ import org.bukkit.entity.EntityType;
+ import org.bukkit.entity.Goat;
++// Paper start
++import org.bukkit.entity.LivingEntity;
++import org.jetbrains.annotations.NotNull;
++// Paper end
+ 
+ public class CraftGoat extends CraftAnimals implements Goat {
+ 
+@@ -34,4 +38,11 @@ public class CraftGoat extends CraftAnimals implements Goat {
+     public void setScreaming(boolean screaming) {
+         this.getHandle().setScreamingGoat(screaming);
+     }
++
++    // Paper start - Goat ram API
++    @Override
++    public void ram(@NotNull LivingEntity entity) {
```

Comment:
```suggestion
+    public void ram(@org.jetbrains.annotations.NotNull org.bukkit.entity.LivingEntity entity) {
```
Sorry, I was a bit inconsistent earlier. For "single use" it's probably better to use the FQN, rather than importing.